src/network/clientpackethandler.cpp

```diff
@@ -1240,28 +1241,130 @@ void Client::handleCommand_HudSetParam(NetworkPacket* pkt)
 
 void Client::handleCommand_HudSetSky(NetworkPacket* pkt)
 {
-	std::string datastring(pkt->getString(0), pkt->getSize());
-	std::istringstream is(datastring, std::ios_base::binary);
+	if (m_proto_ver < 39) {
+		// Handle Protocol 38 and below servers with old set_sky,
+		// ensuring the classic look is kept.
+		std::string datastring(pkt->getString(0), pkt->getSize());
+		std::istringstream is(datastring, std::ios_base::binary);
 
-	video::SColor *bgcolor           = new video::SColor(readARGB8(is));
-	std::string *type                = new std::string(deSerializeString(is));
-	u16 count                        = readU16(is);
-	std::vector<std::string> *params = new std::vector<std::string>;
+		SkyboxParams skybox;
+		skybox.bgcolor = video::SColor(readARGB8(is));
+		skybox.type = std::string(deSerializeString(is));
```

Comment:
What about the other struct variables? They're not initialized properly.

Reply:
Doesn't C++ call the default constructor when you declare a struct?

Reply:
Setting `skybox.tint_type = "default"` doesn't require `sun_tint` or `moon_tint` to be initialised, as sky.cpp handles this special case, (or to use tonemapping when enabled), to use the original sun and moon tinting at dawn/dusk.

Reply:
I should look better at the actual struct variables. Only `clouds` needs to be initialized manually.

Reply:
I've set `sun_tint` and `moon_tint` to `RGBA(255, 255, 255, 255)` just in case *something* goes wrong.